content/appendices.md

```diff
@@ -598,6 +598,13 @@ character set:
 
     extended_user_id_char = %x21-39 / %x3B-7E  ; all ASCII printing chars except :
 
+##### User IDs over federation
+
+Due to a lack of validation in original Matrix homeserver implementations,
+the localpart of user IDs over federation may contain any valid unicode
+codepoints except `:`. A future spec change may create a new room version
+to disallow such user IDs.
```

Comment:
> aaaaaaaargh everything is awful

:+1: 

> I can't see any difference between this and "Historical user IDs". They are both "IDs you're not supposed to create but might have to deal with because apparently validation is hard"

The only difference is that there are real users who have created historical user IDs on normal clients and servers. Arbitrary unicode ids exist in the wild too, but it has never been possible to create them without modifying the server.

> do they even have to be valid unicode codepoints?

I think server implementations try to decode utf-8 sufficiently early so that invalid unicode user IDs explode before being persisted anywhere, but I'm not completely sure.

Reply:
> I think server implementations try to decode utf-8 sufficiently early so that invalid unicode user IDs explode before being persisted anywhere, but I'm not completely sure.

ok, that seems fair. Let's assume they *do* have to be UTF-8 until someone proves otherwise :)